### DISCUSSION TITLE
Debug and Test availability

### DISCUSSION THREAD

**Discussion by hpposch** [elsa-workflows/elsa-core#2919] [Q&A] [answered]
Hi,
i just wanted to ask when do you plan to release the functionalities to debug and test workflows?

Best regards,
Peter

**Comment by sfmskywalker** [comment, marked as answer]
Hi Peter,

There's no concrete plan as of yet to work on workflow debugging.

Current priorities are getting Elsa 3 to have feature-parity with Elsa 2.
After that, effort will probably be put into the Elsa 3 dashboard framework.

Once we got those ducks in a row, it's likely we'll be working on workflow debugging - fully recognizing this is a fundamental aspect of the system.

Elsa 2 currently does have limited support for testing workflows. I don't have documentation to enable this, but here are the steps:

1. Add a reference to the `Elsa.WorkflowTesting.Api` package.
2. Add to Startup: `services.AddWorkflowTestingServices();`

Now you should be able to execute workflows from the designer. For example:

![workflow-testing-from-designer](https://user-images.githubusercontent.com/938393/161974197-2a7b1fd3-f851-4457-9a7c-53f47c2efd27.gif)

**Reply by hpposch** [reply]
Hi
Thank you for you reply!

I am going to try this out, thank you :)

**Comment by hpposch** [comment]
I executed the steps from above and i am able to see the 'Test' Tab now when i open the Designer.
Anyway i do not get any console output (Visual Studio or PowerShell)
Also the green 'Check' Symbol is not here when i click in Execute Workflow.


What am i missing here?

**Reply by benbhall** [reply]
Add the following also:

```
 .UseEndpoints(endpoints =>
    {
        endpoints.MapWorkflowTestHub();

      ...
    });
```

But even then, it doesn't seem to be fully functional as @sfmskywalker suggested. I can't even hit the entry endpoint it gives for an HTTP Endpoint activity. Might be more useful to you in other scenarios in its current state though @hpposch.
